source/modulo_components/modulo_components/component_interface.py

```diff
@@ -49,37 +53,26 @@ def __init__(self, node_name: str, *kargs, **kwargs):
         self._parameter_dict: Dict[str, Union[str, sr.Parameter]] = {}
         self._predicates: Dict[str, Union[bool, Callable[[], bool]]] = {}
         self._predicate_publishers: Dict[str, Publisher] = {}
+        self._periodic_callbacks: Dict[str, Callable[[], None]] = {}
         self.__tf_buffer: Optional[Buffer] = None
         self.__tf_listener: Optional[TransformListener] = None
         self.__tf_broadcaster: Optional[TransformBroadcaster] = None
 
+        self._qos = QoSProfile(depth=10)
```

Comment:
Is this overwriting a base class variable? Because it seems to never been used anywhere so just wondering how it affect the rest.

Reply:
It's going to be used once I add publishers and subscribers in one of my next PRs